RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.cpp

```diff
@@ -0,0 +1,432 @@
+#include "CaloTowerToolFCCee.h"
+
+// FCCSW
+#include "k4Interface/IGeoSvc.h"
+
+// datamodel
+#include "edm4hep/CalorimeterHitCollection.h"
+#include "edm4hep/Cluster.h"
+#include "edm4hep/MutableCluster.h"
+
+// DD4hep
+#include "DD4hep/Detector.h"
+#include "DD4hep/Readout.h"
+
+DECLARE_COMPONENT(CaloTowerToolFCCee)
+
+CaloTowerToolFCCee::CaloTowerToolFCCee(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent), m_geoSvc("GeoSvc", name) {
+  declareProperty("ecalBarrelCells", m_ecalBarrelCells, "");
+  declareProperty("ecalEndcapCells", m_ecalEndcapCells, "");
+  declareProperty("ecalFwdCells", m_ecalFwdCells, "");
+  declareProperty("hcalBarrelCells", m_hcalBarrelCells, "");
+  declareProperty("hcalExtBarrelCells", m_hcalExtBarrelCells, "");
+  declareProperty("hcalEndcapCells", m_hcalEndcapCells, "");
+  declareProperty("hcalFwdCells", m_hcalFwdCells, "");
+  declareInterface<ITowerToolThetaModule>(this);
+}
+
+StatusCode CaloTowerToolFCCee::initialize() {
+  if (GaudiTool::initialize().isFailure()) {
+    return StatusCode::FAILURE;
+  }
+  
+  if (!m_geoSvc) {
+    error() << "Unable to locate Geometry Service. "
+            << "Make sure you have GeoSvc and SimSvc in the right order in the configuration." << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // check if readouts exist & retrieve Module-Theta segmentations
+  // if readout does not exist, reconstruction without this calorimeter part will be performed
+  std::pair<dd4hep::DDSegmentation::Segmentation*, SegmentationType> tmpPair;
+  info() << "Retrieving Ecal barrel segmentation" << endmsg;
+  tmpPair = retrieveSegmentation(m_ecalBarrelReadoutName);
+  m_ecalBarrelSegmentation = tmpPair.first;
+  m_ecalBarrelSegmentationType = tmpPair.second;
+  if (tmpPair.first != nullptr && tmpPair.second == SegmentationType::kWrong) {
+    error() << "Wrong type of segmentation" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_useHalfTower) {
+    m_decoder = m_geoSvc->lcdd()->readout(m_ecalBarrelReadoutName).idSpec().decoder();
```

Comment:
`lcdd()` is deprecated, can you instead use `getDetector()`?

Reply:
Yeah sure, modified.